services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/MapboxV6BaseGeocoding.java

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.api.geocoding.v6;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.mapbox.core.MapboxService;
+
+/**
+ * Base class for entry points to Mapbox V6 geocoding: forward geocoding, reverse geocoding
+ * and batch geocoding. See derived classes for more information.
+ * @param <T> response type.
+ */
+public abstract class MapboxV6BaseGeocoding<T> extends MapboxService<T, V6GeocodingService> {
+
+  @NonNull
+  protected abstract String accessToken();
+
+  @Nullable
+  protected abstract Boolean permanent();
+
+  @Nullable
+  protected abstract String clientAppName();
+
+  @NonNull
+  @Override
+  protected abstract String baseUrl();
+
+  protected MapboxV6BaseGeocoding() {
+    super(V6GeocodingService.class);
+  }
+
+  /**
+   * Base class for Mapbox V6 Geocoding Builders.
+   * @param <T> type of Builder
+   */
+  public abstract static class BaseBuilder<T> {
+
+    protected abstract T accessToken(@NonNull String accessToken);
+
+    /**
+     * Specify whether you intend to store the results of the query. Backend default is false.
+     *
+     * @param permanent specify whether you intend to store the results
+     * @return this builder for chaining options together
+     *
+     * @see <a href="https://docs.mapbox.com/api/search/geocoding-v6/#data-storage">Data storage</a>
+     */
+    public abstract T permanent(@NonNull Boolean permanent);
```

Comment:
Builder functions are public because they're supposed to be called by a user. See [code example](https://github.com/mapbox/mapbox-java/pull/1550/files#diff-1a030938dc097a48285ffcd67724c0fc5d062f0baa16c266f184e5abeedc18d3R72-R77) and this comment for more information

Reply:
Looks like this thread is the same as https://github.com/mapbox/mapbox-java/pull/1550#discussion_r1238347905, let's continue there.